src/components/RouteChangeHandler/RouteChangeHandler.js

```diff
@@ -0,0 +1,110 @@
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router';
+import { ELEMENT_IDS, NO_CATEGORY } from '../../constants';
+
+/**
+ * Add an event listener which only listens once
+ * There is a { once: true } parameter on addEventListener,
+ * but it doesn't seem well-supported ATM
+ */
+const addOneTimeEventListener = (el, event, fn) => {
+  const func = () => {
+    el.removeEventListener(event, func);
+    fn();
+  };
+
+  el.addEventListener(event, func);
+};
+
+/**
+ * If focus point is found, set focus without a visual style until user tabs
+ */
+function focus(oldPath, currentPath) {
+  let focusPoint;
+
+  // first look for a hash
+  if (document.location.hash) {
+    focusPoint = document.querySelector(document.location.hash);
+  }
+
+  // if we have changed cities, and landed on overview
+  // then focus on overview subnav item
+  if (!focusPoint) {
```

Comment:
Minor (and not new in these change) - this is logically just an `else`, right? Might be easier to read like so:
```js
// first look for a hash
if (document.location.hash) {
  focusPoint = document.querySelector(document.location.hash);
} else {
  // if we have changed cities, and landed on overview
  // then focus on overview subnav item
  const oldSegments = oldPath.split('/', 3);
  const currentSegments = currentPath.split('/', 3);

  if (currentSegments[2] === NO_CATEGORY && currentSegments[1] !== oldSegments[1]) {
    focusPoint = document.getElementById(ELEMENT_IDS.OVERVIEW_LINK);
  }
}
```